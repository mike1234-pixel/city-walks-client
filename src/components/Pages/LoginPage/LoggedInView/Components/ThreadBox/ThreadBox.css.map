{"version":3,"sourceRoot":"","sources":["ThreadBox.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;;;AAGF;EACE;EACA;EACA;EACA;EACA;;;AAGF;EACE;;;AAGF;EACE;EACA;EACA","file":"ThreadBox.css"}