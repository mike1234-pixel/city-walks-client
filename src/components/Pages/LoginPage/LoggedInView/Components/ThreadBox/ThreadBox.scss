/** @format */

.thread-box {
  background: var(--linear-blue);
  color: white;
  margin-top: 1rem;
  padding: 30px;
  border-radius: 0.25rem;
}

.thread-container {
  background-color: #2e2e2e;
  padding: 30px;
  border-radius: 0.25rem;
}

.thread-box-title {
  font-family: var(--display-font);
  font-size: 30px;
}

.replies-h3-container {
  margin: 30px 0;
}

.reply {
  background: var(--blue);
  color: black;
  padding: 20px;
  margin: 5px;
  border-radius: 0.25rem;
}

#reply {
  color: white;
}

.show-all-replies-btn {
  margin: 30px 0;
  color: black !important;
  border-color: black !important;
}