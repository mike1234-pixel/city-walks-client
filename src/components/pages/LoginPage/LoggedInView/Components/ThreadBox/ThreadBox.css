/** @format */

#thread-box {
  background-color: var(--elegant-color-dark);
  color: white;
  margin-top: 1rem;
  width: 90vw;
  padding: 30px;
}

.thread-container {
  background-color: #2e2e2e;
  padding: 30px;
}

.thread-box-title {
  font-family: var(--display-font);
  font-size: 50px;
}

.replies-h3-container {
  margin: 30px 0;
}

.reply {
  background: var(--blue);
  color: black;
  padding: 20px;
  margin: 5px;
}

#reply {
  color: white;
}

.show-all-replies-btn {
  margin: 30px 0;
}
