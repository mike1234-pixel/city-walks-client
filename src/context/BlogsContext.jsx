import { createContext, useState } from "react"

export const BlogsContext = createContext();

export const BlogsContextProvider = (props) => {

    const [blogPosts, setBlogPosts] = useState([])
    const [blogsLoading, setBlogsLoading] = useState(true)

    return (
        <BlogsContext.Provider 
            value={{
                blogPosts,
                setBlogPosts,
                blogsLoading,
                setBlogsLoading,
            }}
        >
            {props.children}
        </BlogsContext.Provider>
    )
}