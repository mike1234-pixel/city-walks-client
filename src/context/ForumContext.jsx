import { createContext, useState } from "react"
import BoardBox from "../components/pages/LoginPage/LoggedInView/Components/BoardBox/BoardBox"

export const ForumContext = createContext();

export const ForumContextProvider = (props) => {
 
    const [loadingBoards, setLoadingBoards] = useState(true)
    const [boards, setBoards] = useState([])
    const [selectedThreads, setSelectedThreads] = useState([])

    const [currentBoardName, setCurrentBoardName] = useState("")
    const [currentBoardId, setCurrentBoardId] = useState("")

    let displayBoards;

    if (!loadingBoards) {
    displayBoards = boards.map((board, index) => {
        return (
            <BoardBox boardId={board._id} name={board.name} description={board.description} index={index} key={index} />
                )
        })
        // create routes for boards with Thread as the view component
    }

    return (
        <ForumContext.Provider 
            value={{
                setBoards,
                boards, // data
                setLoadingBoards,
                loadingBoards,
                displayBoards,
                setSelectedThreads,
                selectedThreads, 
                setCurrentBoardName,
                currentBoardName,
                setCurrentBoardId,
                currentBoardId,
            }}>
            {props.children}
        </ForumContext.Provider>
    )
}

