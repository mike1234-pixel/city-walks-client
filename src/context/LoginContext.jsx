import { createContext, useState } from "react"
import axios from "axios"
import qs from "qs"
import LoginForm from "../components/pages/LoginPage/LoginForm/LoginForm";
import RegistrationForm from "../components/pages/LoginPage/RegistrationForm/RegistrationForm"
import VerificationForm from "../components/pages/LoginPage/VerificationForm/VerificationForm";
import ResetPasswordForm from "../components/pages/LoginPage/ResetPasswordForm/ResetPasswordForm";
import ForgotPasswordForm from "../components/pages/LoginPage/ForgotPasswordForm/ForgotPasswordForm";
import Boards from "../components/pages/LoginPage/LoggedInView/Boards"

export const LoginContext = createContext();

export const LoginContextProvider = (props) => {
    const [popupVisible, setPopupVisible] = useState(false)

    const [firstName, setFirstName] = useState("")
    const [lastName, setLastName] = useState("")
    const [userId, setUserId] = useState("")
    const [registrationEmail, setRegistrationEmail] = useState("")
    const [registrationPassword, setRegistrationPassword] = useState("")
  
    const [loginEmail, setLoginEmail] = useState("")
    const [loginPassword, setLoginPassword] = useState("")
  
    // save all of this in local storage
    const [loggedIn, setLoggedIn] = useState(false)
    const [userFirstName, setUserFirstName] = useState("")
    const [userLastName, setUserLastName] = useState("")
  
    const [verificationEmail, setVerificationEmail] = useState("")

    const [resetPasswordEmail, setResetPasswordEmail] = useState("")
    const [resetPasswordOldPassword, setResetPasswordOldPassword] = useState("")
    const [resetPasswordNewPassword, setResetPasswordNewPassword] = useState("")
    const [resetPasswordConfirmNewPassword, setResetPasswordConfirmNewPassword] = useState("")

    const [forgotPasswordEmail, setForgotPasswordEmail] = useState("")

    const handlePopup = () => {
      setPopupVisible(false)
      localStorage.setItem("popupVisible", false)
    }

    // set display form on LoginPage
    const [form, setForm] = useState("loggedOutView");

    let displayForm = <Boards/>;
    if (form === "boards") {
      displayForm = <Boards/>;
    } else if (form === "loginForm") {
      displayForm = <LoginForm />;
    } else if (form === "registrationForm") {
      displayForm = <RegistrationForm />;
    } else if (form === "verificationForm") {
      displayForm = <VerificationForm />;
    } else if (form === "resetPasswordForm") {
      displayForm = <ResetPasswordForm />;
    } else if (form === "forgotPasswordForm") {
      displayForm = <ForgotPasswordForm />;
    }

    const handleChangeRegistration = (event) => {
        switch(event.target.name) {
            case "registration-fname":
              setFirstName(event.target.value)
              break;
            case "registration-lname":
              setLastName(event.target.value)
              break;
            case "registration-email":
              setRegistrationEmail(event.target.value)
              break;
            case "registration-password":
              setRegistrationPassword(event.target.value)
              break;
          } 
    }
  
    const handleSubmitRegistration = (event) => {
      console.log("handle submit triggered")
        event.preventDefault()
  
        const payload = {
            fname: firstName,
            lname: lastName,
            email: registrationEmail,
            password: registrationPassword,
          };
  
    axios
        .post("http://localhost:5000/register-user", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "An account with this email already exists.") {
            alert("An account with this email already exists.")
          } else if (res.data === "We have sent you an email. Please verify your account by clicking the link in the mail.") {
            alert("We have sent you an email. Please verify your account by clicking the link in the email. (This code expires after 10 minutes)")
           } else {
            alert("Registration Complete")
            setLoggedIn(true)
            setFirstName("")
            setLastName("")
            setRegistrationEmail("")
            setRegistrationPassword("")
            setUserFirstName(res.data.fname)
            setUserLastName(res.data.lname)
            window.scrollTo(0, 0)
          }
        });
    }
  
    const handleChangeLogin = (event) => {
      switch(event.target.name) {
          case "login-email":
            setLoginEmail(event.target.value)
            break;
          case "login-password":
            setLoginPassword(event.target.value)
            break;
        } 
  }
  
  const handleSubmitLogin = (event) => {
    console.log("handle submit login triggered")
      event.preventDefault()
  
      const payload = {
          email: loginEmail,
          password: loginPassword
        };
  
  axios
      .post("http://localhost:5000/login-user", qs.stringify(payload))
      .then((res, err) => {
        if (err) {
          console.log(err);
        } else if (res.data === "Your account exists but is not activated. Please click 'verify account' for email verification.") {
          alert("Your account exists but is not activated. Please click 'verify account' for email verification.")
        } else if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Login Complete")
          setLoggedIn(true)
          setLoginEmail("")
          setLoginPassword("")
          setForm("")
          setUserId(res.data._id)
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          localStorage.setItem("loggedIn", true)
          localStorage.setItem("userId", res.data._id)
          localStorage.setItem("userFirstName", res.data.fname)
          window.scrollTo(0, 0)
        }
      });
  }

  // resend verification email

  const handleChangeVerification = (event) => {
    switch(event.target.name) {
        case "verification-email":
          setVerificationEmail(event.target.value)
          break;
      }   
  }

const handleSubmitVerification = (event) => {
    console.log("handle submit verification")
    event.preventDefault()

    const payload = {
        email: verificationEmail,
      };

axios
    .post("http://localhost:5000/reverify-user", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("Verification email sent. Check your inbox.")
        setVerificationEmail("")
        window.scrollTo(0, 0)
      }
    })
}

  // reset password

  const handleChangeResetPassword = (event) => {
    switch(event.target.name) {
        case "reset-email":
          setResetPasswordEmail(event.target.value)
          break;
        case "old-password":
          setResetPasswordOldPassword(event.target.value)
          break;
        case "new-password":
          setResetPasswordNewPassword(event.target.value)
          break;
        case "confirm-new-password":
          setResetPasswordConfirmNewPassword(event.target.value)
          break;
      }   
  }

const handleSubmitResetPassword = (event) => {
    console.log("handle submit reset password")
    event.preventDefault()

    const payload = {
        email: resetPasswordEmail,
        oldPassword: resetPasswordOldPassword,
        newPassword: resetPasswordNewPassword,
      };

      if (resetPasswordNewPassword === resetPasswordConfirmNewPassword) {
        axios
        .post("http://localhost:5000/reset-password-with-old-password", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "user not found") {
            alert("We could not find your account. Please try again.")
          } else if (res.data === "old password does not match password in the database") {
            alert("The password you entered does not match the account. Please try again or use the Forgot Password option.")
          } else if (res.data === "password successfully updated") {
            alert("Password reset. You can now login.")
            setResetPasswordEmail("")
            setResetPasswordOldPassword("")
            setResetPasswordNewPassword("")
            setResetPasswordConfirmNewPassword("")
            window.scrollTo(0, 0)
          }
        })
      } else {
        alert("Passwords don't match. Please try again.")
      }

}

  // forgot password

  const handleChangeForgotPassword = (event) => {
    switch(event.target.name) {
        case "forgot-password-email":
          setForgotPasswordEmail(event.target.value)
          break;
      }   
  }

const handleSubmitForgotPassword = (event) => {
    console.log("handle submit forgot password")
    event.preventDefault()

    const payload = {
        email: forgotPasswordEmail,
      };

axios
    .post("http://localhost:5000/forgot-password", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("We have sent you an email. Please click the click in your email to reset your password")
        setForgotPasswordEmail("")
        window.scrollTo(0, 0)
      }
    })
}
  
  const logOut = () => {
    localStorage.clear()
    localStorage.setItem("popupVisible", false)
    setLoggedIn(false)
    setUserId("")
    setUserFirstName("")
    setUserLastName("")
    setForm("boards")
  }

    return (
        <LoginContext.Provider 
            value={{
              popupVisible,
              setPopupVisible,
              handlePopup,
               // display form
               displayForm,
               setForm,
               // logged in state 
               loggedIn,
               // registration state and functions
                firstName,
                lastName,
                registrationEmail,
                registrationPassword,
                handleChangeRegistration,
                handleSubmitRegistration,
                // login state and functions
                loginEmail,
                loginPassword,
                handleChangeLogin: handleChangeLogin,
                handleSubmitLogin: handleSubmitLogin,
                setLoggedIn,
                setUserFirstName,
                setUserId,
                // logged in user data
                userFirstName,
                userId,
                userLastName,
                // resend verification email
                verificationEmail,
                handleChangeVerification,
                handleSubmitVerification,
                // reset password
                resetPasswordEmail,
                resetPasswordOldPassword,
                resetPasswordNewPassword,
                resetPasswordConfirmNewPassword,
                handleChangeResetPassword,
                handleSubmitResetPassword,
                // forgot password
                forgotPasswordEmail,
                handleChangeForgotPassword,
                handleSubmitForgotPassword,
                // logout function
                logOut
            }}
        >
            {props.children}
        </LoginContext.Provider>
    )
}

// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();

