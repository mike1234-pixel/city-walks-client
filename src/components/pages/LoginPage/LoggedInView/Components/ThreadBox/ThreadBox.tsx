import { ReactNode, useState } from "react"
import React from 'react'
import { MDBInput, MDBBtn, MDBCard, MDBCardTitle, MDBCardText } from "mdbreact"
import axios, { AxiosError, AxiosResponse } from "axios"
import qs from "qs"
import { connect } from 'react-redux'
import Reply from "../../../../../../types/Boards/Reply"
import ReplyToSubmit from '../../../../../../types/PostRequests/Reply'
import SelectedReply from "../../../../../../types/PostRequests/SelectedReply"
import SelectedThread from "../../../../../../types/PostRequests/SelectedThread"
import './ThreadBox.scss'
import GlobalState from "../../../../../../types/State/Global/State"

interface Props {
  currentBoardName: string; 
  threadId: string; 
  userFirstName: string; 
  title: string;
  content: string; 
  replies: Array<Reply>; 
  submittedOn: string; 
  userId: string; 
  loggedIn: string; 
  currentUserFirstName: string; 
  currentUserId: string;
}

const ThreadBox: React.FC<any> = (props: Props) => {

    const { currentBoardName, threadId, userFirstName, title, content, replies, submittedOn, userId, loggedIn, userFirstName: currentUserFirstName, userId: currentUserId } = props

    const [showAllReplies, setShowAllReplies] = useState<boolean>(false)
    const [reply, setReply] = useState<string>("")

    const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
        setReply(event.target.value)
    }

    const handleSubmit: (event: React.FormEvent) => void = (event) => {
        event.preventDefault()
        setReply("")

        let payload: ReplyToSubmit = {
            currentBoardName: currentBoardName,
            threadId: threadId,
            userId: currentUserId,
            userFirstName: currentUserFirstName,
            reply: reply, // send current user id
          };

        axios
        .post("https://city-walks.herokuapp.com/add-reply", qs.stringify(payload))
        .then((res: AxiosResponse) => {
          alert("reply submitted.")
          window.location.reload()
        }).catch((err: AxiosError) => {
          console.log(err)
        });
    }

    const handleDeleteReply: (replyId: string) => void = (replyId) => {

        let payload: SelectedReply = {
            currentBoardName: currentBoardName,
            threadId: threadId,
            replyId: replyId
        };

        axios
        .delete("https://city-walks.herokuapp.com/delete-reply", { data: payload })
        .then((res: AxiosResponse) => {
          alert("reply deleted.")
          window.location.reload()
        }).catch((err: AxiosError) => {
          console.log(err)
        });
    }

    const handleDeleteThread: (threadId: string) => void = (threadId) => {

      let payload: SelectedThread = {
        currentBoardName: currentBoardName,
        threadId: threadId,
      };

      axios
      .delete("https://city-walks.herokuapp.com/delete-thread", { data: payload })
      .then((res: AxiosResponse) => {

          alert("thread deleted.")
          window.location.reload()
        
      }).catch((err: AxiosError) => {
        console.log(err)
      });
    }

    const replyComponents: Array<ReactNode> = replies.map((reply: Reply, index: number) => {

        return (
        <div className="reply" key={index}>
            <p>{reply.userFirstName} replied!</p>
            <p>{reply.reply}</p>
            <p>{reply.submittedOn.replace('T', ' ').substring(0, 19)}</p>
            {reply.userId === currentUserId && <MDBBtn outline onClick={() => handleDeleteReply(reply._id)}>Delete</MDBBtn>}
        </div>
        )
    })

    let displayReplies: Array<ReactNode>;
    if (showAllReplies) {
        displayReplies = replyComponents;
    } else {
        // if end is greater than array length, slice uses the array length as the last index
        displayReplies = replyComponents.slice(0,3)
    }

    return (
        <MDBCard className="card-body" id="thread-box">
          <div className="thread-container">
            <MDBCardTitle className="thread-box-title">{title}</MDBCardTitle>
                <MDBCardText className="thread-box-content">
                    {content}
                </MDBCardText>
                <p>Posted by {userFirstName} on {submittedOn.replace('T', ' ').substring(0, 19)}</p>
                  {userId === currentUserId && <MDBBtn outline color="danger" onClick={() => handleDeleteThread(threadId)}>Delete Thread</MDBBtn>}
            </div>
            <div className="replies-h3-container">
                <h3>Replies...</h3>
            </div>
                <div className="replies-container">{displayReplies}</div>
                <MDBBtn outline color="default" className="show-all-replies-btn" onClick={() => setShowAllReplies(!showAllReplies)}>{showAllReplies ? "hide replies" : "show all replies"}</MDBBtn>
                <form onSubmit={handleSubmit}>
                    {loggedIn && <div><MDBInput type="textarea" rows="5" name="reply" id="reply" label="reply" value={reply} onChange={handleChange}/><MDBBtn outline type="submit">Submit</MDBBtn></div>  }
                </form>
        </MDBCard>
    )
}

const mapStateToProps = (state: GlobalState) => ({
  loggedIn: state.loginState.loggedIn,
  userFirstName: state.loginState.userFirstName,
  userId: state.loginState.userId,
});

export default connect(mapStateToProps)(ThreadBox);


// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();