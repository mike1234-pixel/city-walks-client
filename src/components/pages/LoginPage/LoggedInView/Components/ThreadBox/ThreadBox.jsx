import { useState, useContext } from "react"
import { MDBInput, MDBBtn, MDBCard, MDBCardTitle, MDBCardText } from "mdbreact"
import axios from "axios"
import qs from "qs"
import { LoginContext } from "../../../../../../context/LoginContext"
import './ThreadBox.css'

const ThreadBox = (props) => {

    const { userFirstName: currentUserFirstName, userId: currentUserId, loggedIn } = useContext(LoginContext)

    const { currentBoardName, threadId, userFirstName, title, content, replies, submittedOn, userId } = props

    const [showAllReplies, setShowAllReplies] = useState(false)
    const [reply, setReply] = useState("")

    const handleChange = (event) => {
        setReply(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()
        setReply("")

        let payload = {
            currentBoardName: currentBoardName,
            threadId: threadId,
            userId: currentUserId,
            userFirstName: currentUserFirstName,
            reply: reply, // send current user id
          };

        axios
        .post("http://localhost:5000/add-reply", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else {
              alert("reply submitted.")
              window.location.reload()
          }
        });
    }

    const handleDeleteReply = (replyId) => {
        let payload = {
            currentBoardName: currentBoardName,
            threadId: threadId,
            replyId: replyId
          };

        axios
        .delete("http://localhost:5000/delete-reply", { data: payload })
        .then((res, err) => {
          if (err) {
            console.log(err);
          }  else {
            alert("reply deleted.")
            window.location.reload()
          }
        });
    }

    const handleDeleteThread = (threadId) => {
      console.log(threadId)

      let payload = {
        currentBoardName: currentBoardName,
        threadId: threadId,
      };

      axios
      .delete("http://localhost:5000/delete-thread", { data: payload })
      .then((res, err) => {
        if (err) {
          console.log(err);
        }  else {
          alert("thread deleted.")
          window.location.reload()
        }
      });
    }

    const replyComponents = replies.map((reply, index) => {
        return (
        <div className="reply" key={index}>
            <p>{reply.userFirstName} replied!</p>
            <p>{reply.reply}</p>
            <p>{reply.submittedOn.replace('T', ' ').substring(0, 19)}</p>
            {reply.userId === currentUserId && <MDBBtn outline color="black" onClick={() => handleDeleteReply(reply._id)}>Delete</MDBBtn>}
        </div>
        )
    })

    let displayReplies;
    if (showAllReplies) {
        displayReplies = replyComponents;
        console.log(replyComponents)
    } else {
        // if end is greater than array length, slice uses the array length as the last index
        displayReplies = replyComponents.slice(0,3)
    }

    return (
        <MDBCard className="card-body" id="thread-box">
          <div className="thread-container">
            <MDBCardTitle className="thread-box-title">{title}</MDBCardTitle>
                <MDBCardText className="thread-box-content">
                    {content}
                </MDBCardText>
                <p>Posted by {userFirstName} on {submittedOn.replace('T', ' ').substring(0, 19)}</p>
                  {userId === currentUserId && <MDBBtn outline color="danger" onClick={() => handleDeleteThread(threadId)}>Delete Thread</MDBBtn>}
            </div>
            <div className="replies-h3-container">
                <h3>Replies...</h3>
            </div>
                <div className="replies-container">{displayReplies}</div>
                <MDBBtn outline color="default" className="show-all-replies-btn" onClick={() => setShowAllReplies(!showAllReplies)}>{showAllReplies ? "hide replies" : "show all replies"}</MDBBtn>
                <form onSubmit={handleSubmit}>
                    {loggedIn && <div><MDBInput type="textarea" rows="5" name="reply" id="reply" label="reply" value={reply} onChange={handleChange}/><MDBBtn outline color="white" type="submit">Submit</MDBBtn></div>  }
                </form>
        </MDBCard>
    )
}

export default ThreadBox

// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();