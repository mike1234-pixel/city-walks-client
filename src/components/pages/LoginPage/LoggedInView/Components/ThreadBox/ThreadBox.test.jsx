import renderer from 'react-test-renderer'
import { LoginContextProvider } from "../../../../../../context/LoginContext"
import ThreadBox from "./ThreadBox"

// snapshot test
it('ThreadBox renders correctly', () => {  

    const currentBoardName = "Your City"
    const threadId = "dvsbhjvsdkjdskjsdvjsd"
    const userFirstName = "John"
    const title = "Have you checked out this great restaurant?"
    const content = "There's a great new restaurant just opened in Sheffield..."
    const replies = [{
        "userFirstName": "James",
        "reply": "ive been there its cool",
        "submittedOn": "2021-01-13T18:08:18Z"
    },
    {
        "userFirstName": "Baz",
        "reply": "it's lame",
        "submittedOn": "2021-01-13T18:08:18Z"
    }]
    const submittedOn = "2021-01-13T18:08:18Z"
    const userId = "vdkjbdjjnsdn"

  const tree = renderer
    .create(
                <LoginContextProvider>
                    <ThreadBox 
                        currentBoardName={currentBoardName}
                        threadId={threadId}
                        userFirstName={userFirstName}
                        title={title}
                        content={content}
                        replies={replies}
                        submittedOn={submittedOn}
                        userId={userId}
                        />
                </LoginContextProvider>
            )
    .toJSON();
  expect(tree).toMatchSnapshot();
});