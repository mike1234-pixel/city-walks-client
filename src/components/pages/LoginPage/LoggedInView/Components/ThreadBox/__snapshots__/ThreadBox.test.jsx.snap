// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadBox renders correctly 1`] = `
<div
  className="card card-body"
  data-test="card"
  id="thread-box"
>
  <div
    className="thread-container"
  >
    <h4
      className="card-title thread-box-title"
      data-test="card-title"
    >
      Have you checked out this great restaurant?
    </h4>
    <p
      className="card-text thread-box-content"
      data-test="card-text"
    >
      There's a great new restaurant just opened in Sheffield...
    </p>
    <p>
      Posted by 
      John
       on 
      2021-01-13 18:08:18
    </p>
  </div>
  <div
    className="replies-h3-container"
  >
    <h3>
      Replies...
    </h3>
  </div>
  <div
    className="replies-container"
  >
    <div
      className="reply"
    >
      <p>
        James
         replied!
      </p>
      <p>
        ive been there its cool
      </p>
      <p>
        2021-01-13 18:08:18
      </p>
    </div>
    <div
      className="reply"
    >
      <p>
        Baz
         replied!
      </p>
      <p>
        it's lame
      </p>
      <p>
        2021-01-13 18:08:18
      </p>
    </div>
  </div>
  <button
    className="btn-default btn-outline-default btn Ripple-parent show-all-replies-btn"
    data-test="button"
    onClick={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    show all replies
    <div
      className="Ripple Ripple-outline "
      data-test="waves"
      style={
        Object {
          "height": "0px",
          "left": "0px",
          "top": "0px",
          "width": "0px",
        }
      }
    />
  </button>
  <form
    onSubmit={[Function]}
  />
</div>
`;
